lib/modules/datasource/hexpm-bob/index.ts

```diff
@@ -0,0 +1,180 @@
+import { logger } from '../../../logger';
+import { ExternalHostError } from '../../../types/errors/external-host-error';
+import { cache } from '../../../util/cache/package/decorator';
+import { HttpError } from '../../../util/http';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import { PackageType, datasource, defaultRegistryUrl } from './common';
+
+export class HexpmBobDatasource extends Datasource {
+  static readonly id = datasource;
+
+  constructor() {
+    super(datasource);
+  }
+
+  override readonly customRegistrySupport = true;
+
+  override readonly defaultRegistryUrls = [defaultRegistryUrl];
+
+  override readonly caching = true;
+
+  @cache({
+    namespace: `datasource-${datasource}`,
+    key: ({ registryUrl, packageName }: GetReleasesConfig) =>
+      `${registryUrl ?? defaultRegistryUrl}:${packageName}`,
+  })
+  async getReleases({
+    registryUrl,
+    packageName,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    const packageType = this.getPackageType(packageName);
+
+    if (!packageType) {
+      return null;
+    }
+
+    logger.trace(
+      { registryUrl: registryUrl ?? defaultRegistryUrl, packageName },
+      `fetching hex.pm bob ${packageName} release`
+    );
+
+    const url = `${
+      registryUrl ?? defaultRegistryUrl
+    }/builds/${packageName}/builds.txt`;
+
+    const result: ReleaseResult = {
+      releases: [],
+      ...this.getPackageDetails(packageType),
+    };
+    try {
+      const { body } = await this.http.get(url);
+      result.releases = body
+        .split('\n')
+        .map((line) => line.trim())
+        .filter((line) => line !== '')
+        .map((line) => {
+          const [version, gitRef, buildDate, newDigest] = line.split(' ');
+
+          return {
+            changelogUrl: this.changelogUrl(version, packageType),
+            downloadUrl: `${
+              registryUrl ?? defaultRegistryUrl
+            }builds/${packageName}/${version}.tar.gz`,
+            gitRef,
+            isStable: this.isStable(version, packageType),
+            releaseTimestamp: new Date(buildDate).getTime(),
+            version: this.cleanVersion(version, packageType),
+            newDigest,
+            constraints: this.constraints(version, packageType),
+            sourceUrl: this.sourceUrl(gitRef, packageType),
+          };
+        });
+    } catch (err) {
+      if (err instanceof HttpError && err.response?.statusCode !== 404) {
+        throw new ExternalHostError(err);
+      }
+      this.handleGenericErrors(err);
+    }
+
+    return result.releases.length > 0 ? result : null;
+  }
+
+  private getPackageType(packageName: string): PackageType | null {
+    if (packageName === 'elixir') {
+      return 'elixir';
+    }
+    if (packageName.startsWith('otp/')) {
+      return 'erlang';
+    }
+    return null;
+  }
+
+  // eslint-disable-next-line consistent-return
```

Comment:
is that really required ?

Reply:
@viceice How should I proceed on this issue?

Reply:
@rarkins @zharinov @JamieMagee Should we disable via eslint config?

@maennchen I think we can disable that rule globally. So please add a new PR to be merged first.

Reply:
I don't think we need, let's provide default return instead

Reply:
`consistent-return` is a js eslint rule, which doesn't know that the default return can't be hit. Also typescript will know never reachable code and will warn about that. so we need to disable `consistent-return` here.

Reply:
Personally, I would just write `version.replace(/^(?:v|OTP-)/, '')` in this scenario if this is correct enough here.

Reply:
> Also typescript will know never reachable code and will warn about that

I doubt TypeScript will warn about incomprehensive switch clause, in the case if `PackageType` will be extended with new values some day.

Reply:
Actually I don't care too much about this eslint rule, you're welcome to remove it. Just consider whether it makes sense to get rid of `PackageType` and apply both types of transformations for every package (i.e. without `switch`-ing). This can simplify code branches and can help to avoid writing ridiculous test cases just for making our code coverage checks pass.

Reply:
@zharinov 

Typescript recognizes it:


```diff
diff --git a/lib/modules/datasource/hexpm-bob/index.ts b/lib/modules/datasource/hexpm-bob/index.ts
index c0f883fcc..1d49ae34c 100644
--- a/lib/modules/datasource/hexpm-bob/index.ts
+++ b/lib/modules/datasource/hexpm-bob/index.ts
@@ -99,8 +99,8 @@ export class HexpmBobDatasource extends Datasource {
     switch (packageType) {
       case 'elixir':
         return version.match(/^v\d+\.\d+\.\d+($|-otp)/) !== null;
-      case 'erlang':
-        return version.startsWith('OTP-');
+      // case 'erlang':
+      //   return version.startsWith('OTP-');
     }
   }
```

=>

```
lib/modules/datasource/hexpm-bob/index.ts:98:64 - error TS2366: Function lacks ending return statement and return type does not include 'undefined'.

98   private isStable(version: string, packageType: PackageType): boolean {
                                                                  ~~~~~~~


Found 1 error in lib/modules/datasource/hexpm-bob/index.ts:98
```

Reply:
Now I see. Please, remove the obsolete eslint rule 👍

Reply:
https://github.com/renovatebot/renovate/pull/19101